Product/Production/Gateway/ExchangeManagement/src/main/java/gov/hhs/fha/nhinc/exchangemanagement/ExchangeManagement.java

```diff
@@ -0,0 +1,161 @@
+import gov.hhs.fha.nhinc.exchangemgr.util.ExchangeManagerUtil;
+import org.apache.commons.lang.StringUtils;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ *
+ * @author ttang
+ */
+public class ExchangeManagement implements EntityExchangeManagementPortType {
+
+    private static final Logger LOG = LoggerFactory.getLogger(ExchangeManagement.class);
+    private final ExchangeManager exchangeManager = ExchangeManager.getInstance();
+    private static final String ACT_SUCCESSFUL = "successful";
+    private static final String ACT_FAIL = "fail";
+    private static final String ACT_BUSY = "refresh locked";
+
+    @Override
+    public SimpleExchangeManagementResponseMessageType deleteExchange(DeleteExchangeRequestMessageType arg0) {
+        SimpleExchangeManagementResponseMessageType response = newSimpleResponse();
+        String exchangeName = arg0.getExchangeName();
+
+        if (StringUtils.isBlank(exchangeName)) {
+            response.setMessage("exchange-name cannot be blank");
+            return response;
+        }
+
+        if (exchangeManager.isRefreshLocked()) {
+            response.setMessage(ACT_BUSY);
+            return response;
+        }
+
+        try {
+            response.setStatus(exchangeManager.deleteExchange(exchangeName));
+            response.setMessage(ACT_SUCCESSFUL);
+        } catch (ExchangeManagerException e) {
+            response.setMessage(ACT_FAIL);
+            LOG.error("error during delete-exchange: {}", e.getLocalizedMessage(), e);
+        }
+        return response;
+    }
+
+    @Override
+    public SimpleExchangeManagementResponseMessageType saveExchange(SaveExchangeRequestMessageType arg0) {
+        SimpleExchangeManagementResponseMessageType response = newSimpleResponse();
+
+        if (exchangeManager.isRefreshLocked()) {
+            response.setMessage(ACT_BUSY);
+            return response;
+        }
+
+        try {
+            response.setStatus(exchangeManager.saveExchange(arg0.getExchange()));
+            response.setMessage(ACT_SUCCESSFUL);
+        } catch (ExchangeManagerException e) {
+            response.setMessage(ACT_FAIL);
+            LOG.error("error during save-exchange: {}", e.getLocalizedMessage(), e);
+        }
+
+        return response;
+    }
+
+    @Override
+    public RefreshExchangeManagerResponseMessageType refreshExchangeManager(
+        RefreshExchangeManagerRequestMessageType arg0) {
+        RefreshExchangeManagerResponseMessageType response = new RefreshExchangeManagerResponseMessageType();
+
+        if (exchangeManager.isRefreshLocked()) {
+            LOG.info("exchange-manager is busy: {}", ACT_BUSY);
+        } else {
+            response.getExchangeDownloadStatusList().addAll(ExchangeManagerUtil.forceExchangesRefresh());
+        }
+        return response;
+    }
+
+    @Override
+    public GetExchangeInfoViewResponseMessageType getExchangeInfoView(GetExchangeInfoViewRequestMessageType arg0) {
```

Comment:
Why do we have all these stubbed methods?

Reply:
these method are created for the next tickets